nest-city-account/src/utils/subservices/line-logger.subservice.ts

```diff
@@ -26,73 +27,83 @@ export class LineLoggerSubservice implements LoggerService {
     this.color = color
   }
 
+  private formatStringMessage(messages: string): string {
+    if (messages.length === 0) return ''
+    return isLogfmt(messages)
+      ? ' '.concat(messages)
+      : `message="${escapeForLogfmt(messages)}"`
+  }
+
   private printLog(
     severity: string,
     message: unknown,
     optionalParams: unknown[],
-    colorCode: string
-  ) {
+    colorCode: string,
+  ): void {
     const completeArray = [message, ...optionalParams]
 
-    let stringItems = ''
-    const otherItems: unknown[] = []
-
-    completeArray.forEach((item) => {
-      if (typeof item === 'string') {
-        stringItems += item + ' '
-      } else {
-        otherItems.push(item)
-      }
-      stringItems = stringItems.trim()
-    })
-
-    let logfmtStringItems = ''
-    if (stringItems.length != 0) {
-      if (isLogfmt(stringItems)) {
-        logfmtStringItems = " ".concat(stringItems)
-      }
-      else {
-        logfmtStringItems = ` message="${escapeForLogfmt(stringItems)}"`
-      }
-    }
-
-    const logfmtOtherItems = otherItems.map((param) => ToLogfmt(param)).join(' ')
-
-    let context = ''
-    if (this.context) {
-      context = ` context="${this.context}"`
-    }
+    const stringMessages = completeArray
+      .filter((item): item is string => typeof item === 'string')
+      .join(' ')
+
+    const otherItems = completeArray.filter(
+      (item): item is string => typeof item !== 'string',
+    )
+
+    const formattedStringMessages = this.formatStringMessage(stringMessages)
+
+    const formattedOtherItems = otherItems
+      .map((item) => ToLogfmt(item))
+      .join(' ')
+
+    const formattedContext = this.context ? `context="${this.context}"` : ''
+
+    const colorStart = this.color ? colorCode : ''
+    const colorEnd = this.color ? '\u001B[0m' : ''
```

Comment:
Would be really cool, of this code was correctly named, such as on a separate line before the color set, such as 
`const ANSI_RESET = '\u001B[0m'`

If it is wrapped under the color end logic in the ternary operator, it's much harder to read, as if it is a constant for some ansi code, you don't need to question it and can read it easily in the operator.